src/lib/js/parse_cms_models/cards.js

```diff
@@ -20,19 +20,50 @@ export function blog_posts(post) {
     content_creators: post.content_creators,
   };
 }
+
+/**
+ * Extracts local chapter names from a raw subelement array and adds them to the parsed element.
+ *
+ * @param {object} parsedElement - The element to which the local chapter names should be added.
+ * @param {Array} lcSbuElementsRaw - The raw subelement array.
+ * @param {string} entryName - The name of the entry in the parsed element.
+ *   Defaults to 'correlaidx'.
+ */
+function parseLcSubElements(
+  parsedElement,
+  lcSbuElementsRaw,
+  entryName = 'correlaidx',
+) {
+  if (lcSbuElementsRaw.length > 0) {
+    parsedElement[entryName] = lcSbuElementsRaw.map((lc) => {
+      if (typeof lc.Local_Chapters_id.translations[0].city !== 'string') {
+        throw new Error('Local chapter name is missing or not a string');
+      }
+      return lc.Local_Chapters_id.translations[0].city;
+    });
+  } else if (lcSbuElementsRaw.length === 0) {
+    parsedElement[entryName] = lcSbuElementsRaw.map((lc) => {
+      return [];
+    });
```

Comment:
@jstet Do we need this clause? I left it out on purpose because not having the key `entryName` represents the absence of local chapters just as much as `{ entryName: [] }`, whereas the first option allows the usage of default parameter values for the components such as `Project_Card` and `Event_Card`.

If there is a good reason to keep it, we should still simplify. Right now it unnecessarily maps across an empty list, which always results in `[]` independent of the callback. Explicit setting would be more readable and get rid of the superflous callback. 

```suggestion
  } else if (lcSbuElementsRaw.length === 0) {
    parsedElement[entryName] = [];
```

Reply:
Without it, the filter did not work because an empty list for lcs means that its a global event. without this clause, the lcs object was not present in the returned data at all when the event had no lc.
Your suggested change looks better though yes